OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -1049,11 +971,24 @@ public static void provideUserConsent(boolean consent) {
       saveUserConsentStatus(consent);
 
       if (!previousConsentStatus && consent && delayedInitParams != null) {
-         OneSignal.init(delayedInitParams.context, delayedInitParams.googleProjectNumber, delayedInitParams.appId, delayedInitParams.openedHandler, delayedInitParams.receivedHandler);
-         delayedInitParams = null;
+         OneSignal.Log(LOG_LEVEL.VERBOSE, "Privacy consent provided, reassigning all delayed init params and attempting init again...");
+         reassignDelayedInitParams();
       }
    }
 
+   private static void reassignDelayedInitParams() {
+      appContext = delayedInitParams.context;
+      googleProjectNumber = delayedInitParams.googleProjectNumber;
+      appId = delayedInitParams.appId;
+      notificationWillShowInForegroundHandler = delayedInitParams.notificationWillShowInForegroundHandler;
+      notificationOpenedHandler = delayedInitParams.notificationOpenedHandler;
+
+      delayedInitParams = null;
+
+       if (appContext != null)
```

Comment:
Seems like a hack of setting the same value. We should call the method we need to here.